lib/Serialization/ModuleFormat.h

```diff
@@ -1703,6 +1703,12 @@ namespace decls_block {
     BCVBR<5>  // index
   >;
 
+  using XRefClangTemplateInstantiationLayout = BCRecordLayout<
+    XREF_CLANG_TEMPLATE_INSTANTIATION,
+    IdentifierIDField, // template name
+    BCArray<IdentifierIDField> // template args
```

Comment:
Template args are not necessarily single identifiers -- the types may be nested in namespaces, may be template specializations themselves, and even can be non-type template parameters.

Is there an existing way to serialize a Clang type?